Mta 4707 (#3300)
scaleway/docs-content: tutorials/easydeploy-gitlab-runner/index.mdx

6. Select the **GitLab Runner** application.
    <Message type="tip">
        If you cannot find GitLab Runner on the first page, use the search bar or navigate through the library using the arrow buttons.
    </Message>
7. Optionally, customize the default configuration for GitLab Runner using [Helm Charts](/tutorials/kubernetes-package-management-helm/). If you do not require any customized configuration you can skip this step.
8. Enter a name (e.g. `gitlab-runner`) and a Kubernetes namespace for your application. If no name is entered, GitLab Runner will be installed in the default namespace of the cluster.
9. Click **Deploy Application** to deploy GitLab Runner on your Kubernetes cluster.
10. Verify that the GitLab Runner is successfully installed and running:
    ```bash
    kubectl get pods -n default # replace "default" with the name of the Kubernetes namespace in which you have installed your GitLab Runner.
    ```
    You should see a pod with a name similar to `gitlab-runner-xxxxxx-xxxxx` in the `Running` state.

## Configuring a GitLab CI/CD pipeline to use your Kubernetes Runner

<Message type="tip">

>   In case you do not have a GitLab repository yet, you can deploy a [GitLab server using Easy Deploy](/tutorials/gitlab-server/) on your Kubernetes cluster.

Reviewer: ```suggestion
  If you do not have a GitLab repository yet, you can deploy a [GitLab server using Easy Deploy](/tutorials/gitlab-server/) on your Kubernetes cluster.
```

2. Register the GitLab Runner:
   - Connect to your Kubernetes cluster where the GitLab Runner is deployed.
   - Use the following command to start the registration process:
     ```bash
     kubectl exec -it <gitlab-runner-pod-name> -- gitlab-runner register
     ```
   - When prompted, enter the following details:
     - **GitLab instance URL:** The URL of your GitLab server (e.g., `https://gitlab.com` or your self-hosted GitLab URL).
     - **Registration Token:** The token you obtained in the previous step.
     - **Description:** A description for this runner (e.g., `k8s-runner`).
     - **Tags:** Tags for the runner (e.g., `k8s-runner`). These tags should match the tags specified in your `.gitlab-ci.yml` file.
     - **Executor:** Select `kubernetes` as the executor.

### Verify Runner registration

After registering the runner, verify that it is correctly registered and active:

1. In your GitLab project, navigate to **Settings** > **CI/CD** > **Runners**.
2. You should see your newly registered runner listed under the **Specific Runners** section.

### Configure the GitLab CI/CD pipeline

1. Create/Edit the `.gitlab-ci.yml` file:

>     - Ensure your `.gitlab-ci.yml` file includes jobs with tags that match your runner’s tags. Here is an example configuration:

Reviewer 2: ```suggestion
    - Ensure your `.gitlab-ci.yml` file includes jobs with tags that match your runner’s tags. Refer to the configuration example below to picture what your file should look like:
```

    ```yaml
    stages:
      - build
      - test

    build:
      stage: build
      script:
        - echo "Building the project..."
        - # Add your build steps here
      tags:
        - k8s-runner

    test:
      stage: test
      script:
        - echo "Running tests..."
        - # Add your test steps here
      tags:
        - k8s-runner
    ```

> 2. Push the Configuration to GitLab:

Reviewer: ```suggestion
2. Push the configuration to GitLab:
```

> 2. Push the Configuration to GitLab:

Reviewer 2: ```suggestion
2. Push the configuration to GitLab:
```

4. Click **Deploy Application**. The application deployment wizard displays.
5. Choose **Application Library** to see the list of available applications.
6. Select the **GitLab Runner** application.
    <Message type="tip">
        If you cannot find GitLab Runner on the first page, use the search bar or navigate through the library using the arrow buttons.
    </Message>
7. Optionally, customize the default configuration for GitLab Runner using [Helm Charts](/tutorials/kubernetes-package-management-helm/). If you do not require any customized configuration you can skip this step.
8. Enter a name (e.g. `gitlab-runner`) and a Kubernetes namespace for your application. If no name is entered, GitLab Runner will be installed in the default namespace of the cluster.
9. Click **Deploy Application** to deploy GitLab Runner on your Kubernetes cluster.
10. Verify that the GitLab Runner is successfully installed and running:
    ```bash
    kubectl get pods -n default # replace "default" with the name of the Kubernetes namespace in which you have installed your GitLab Runner.
    ```
    You should see a pod with a name similar to `gitlab-runner-xxxxxx-xxxxx` in the `Running` state.

> ## Configuring a GitLab CI/CD pipeline to use your Kubernetes Runner

Reviewer 3: I was not able to go beyond this step.
Where is the gitlab repository? (not a pre-requisite)
How is it connected to my gitlab-runner in my k8s cluster?

Author: Added more steps
